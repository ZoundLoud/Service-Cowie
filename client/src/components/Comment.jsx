import React from 'react';
import styled from 'styled-components';

class Comment extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      display: 0,
      commentCount: 0,
    };
    this.handleHoverChange = this.handleHoverChange.bind(this);
    this.handleLeave = this.handleLeave.bind(this);
  }

  componentDidMount() {
    const randomNum = () => Math.floor(Math.random() * 10);
    this.setState({
      commentCount: randomNum(),
    });
  }

  handleHoverChange(index) {
    this.setState({
      display: index,
    });
  }

  handleLeave() {
    this.setState({
      display: 0,
    });
  }

  render() {
    const CommentImage = () => {
      const imgArray = [];
      for (let i = 1; i < this.state.commentCount; i += 1) {
        imgArray.push(
          <CommentDiv
            key={i}
            style={{ backgroundImage: `url(${this.props.random[i]})` }}
            onMouseEnter={() => this.handleHoverChange(i)}
            onMouseLeave={() => this.handleLeave()}
            isHovered={Boolean(this.state.display) >= i}
          >
            <CommentSpan isHovered={Boolean(this.state.display) >= i}>
              {this.props.info.comment}
            </CommentSpan>
            <NameSpan isHovered={Boolean(this.state.display) >= i}>
              {this.props.info.artistname}
            </NameSpan>
          </CommentDiv>,
        );
      }
      return imgArray;
    };
    return (
      <div className="comment-container">
        {CommentImage()}
      </div>
    );
  }
}

const CommentDiv = styled.div`
  width: 22px;
  height: 22px;
  justify-content: space-between;
`;

const CommentSpan = styled.span`
  color: white;
  font-size: 10px;
  position: absolute;
  right: 77%
  top: 100%;
  margin-top: 10px;
  margin-left: 34px;
  visibility: ${props => (props.isHovered ? 'visible' : 'hidden')}
`;

const NameSpan = styled.span`
  color: #f50;
  font-size: 10px;
  position: absolute;
  top: 100%;
  margin-top: 10px;
  visibility: ${props => (props.isHovered ? 'visible' : 'hidden')}
`;

// visibility: ${props => (props.isHovered ? 'visible' : 'hidden')}

export default Comment;


// add text / username from random data
// if hover is true color will = black if false color will be clear or whatever
// can add conditional for timestamp as well

// to get random number of comments.  generate a random number between 1-22 and put it in place of i < blank on span tage generation.  try it at 100 and see how it looks.  maybe they naturally overlap
