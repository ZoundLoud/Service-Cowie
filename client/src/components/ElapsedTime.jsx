import React from 'react';

const ElapsedTime = props => (
  <div className="Time-Container">
    <div className="Elapsed-time">
        2 Years Ago
    </div>
    <button className="hashtag" type="submit"># COWIE</button>
  </div>
);

export default ElapsedTime;
