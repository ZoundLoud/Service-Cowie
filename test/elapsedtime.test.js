import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount, shallow } from 'enzyme';
import Comment from '../client/src/components/Comment';
import ElapsedTime from '../client/src/components/ElapsedTime';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('ElapsedTime Component', () => {
  it('should render ElapsedTime without throwing an error', () => {
    const props = {
      timelapsed: '2018-04-07 00:38:13',
      hashtag: 'SCSI',
    }
    const elapsedtime = shallow(<ElapsedTime info={props.timelapsed} />);
    expect(toJson(elapsedtime)).toMatchSnapshot();
  });
  it('should render correctly in "debug" mode', () => {
    const props = {
      timelapsed: '2018-04-07 00:38:13',
      hashtag: 'SCSI',
    }
    const debug = shallow(<ElapsedTime info={props.timelapsed} debug />);
  
    expect(toJson(debug)).toMatchSnapshot();
  });
});
