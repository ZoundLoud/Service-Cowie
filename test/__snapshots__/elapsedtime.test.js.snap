// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElapsedTime Component should render ElapsedTime without throwing an error 1`] = `
<div
  className="Time-Container"
>
  <div
    className="Elapsed-time"
  >
    a few seconds ago
  </div>
  <button
    className="hashtag"
    type="submit"
  >
    # 
  </button>
</div>
`;

exports[`ElapsedTime Component should render correctly in "debug" mode 1`] = `
<div
  className="Time-Container"
>
  <div
    className="Elapsed-time"
  >
    a few seconds ago
  </div>
  <button
    className="hashtag"
    type="submit"
  >
    # 
  </button>
</div>
`;
